File: ref/tree/binary_search_tree.py
class Node:
   def __init__(self, data):
      self.left = None
      self.right = None
      self.data = data

def InorderTraversal(root):
   if root:
      InorderTraversal(root.leftChild)
      print(root.data)
      InorderTraversal(root.rightChild)

# Insert method to create nodes
   def insert(self, data):
      if self.data:
         if data < self.data:
            if self.left is None:
               self.left = Node(data)
            else:
               self.left.insert(data)
         elif data > self.data:
            if self.right is None:
               self.right = Node(data)
            else:
               self.right.insert(data)
         else:
            self.data = data
# search method to compare the value with nodes
   def search(self, key):
      if key < self.data:
         if self.left is None:
            return str(key)+" Not Found"
         return self.left.search(key)
      elif key > self.data:
         if self.right is None:
            return str(key)+" Not Found"
         return self.right.search(key)
      else:
         print(str(self.data) + ' is found')

File: ref/tree/binary_search_tree.js
"use strict";
class TreeNode {
    constructor(val, left, right) {
        this.val = (val === undefined ? 0 : val);
        this.left = (left === undefined ? null : left);
        this.right = (right === undefined ? null : right);
    }
}
function isSameTree(p, q) {
    /*  Given the roots of two binary trees p and q:
        write a function to check if they are the same or not.
        Two binary trees are considered the same if they are:
        1. structurally identical
        2. and the nodes have the same value.
    */
    // Initial idea: Pick any binary tree traversal search
    // Traverse both at the same time looking for absolute equality
    // Go left then right making sure both are true
    // Can check this node first
    if (p === null || q === null) {
        return p === q;
    }
    else if (p.val === q.val) {
        if (isSameTree(p.right, q.right)) {
            if (isSameTree(p.left, q.left)) {
                return true;
            }
        }
    }
    return false;
}
;
